Anagram/Anagram.cs

```diff
@@ -0,0 +1,77 @@
+﻿
+namespace Anagram
+{
+    public partial class Anagram : Form
+    {
+        public Anagram()
+        {
+            InitializeComponent();
+        }
+        private void Anagrams_Click(object sender, EventArgs e)
+        { 
+                if (inputStringBox.Text == "")
+                {
+                    MessageBox.Show("Please input string first!");
+                }
+
+                String[] wordList = System.IO.File.ReadAllText(@"C:\Users\Krasskoo\Desktop\WordList.txt").Split(new char[] { '.', '?', '!', ' ', ';', ':', ',', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+                int totalWords = wordList.Length;
+
+                String inputString = inputStringBox.Text;
+ 
+                List<string> matchedWordsListFinal = new List<string>();
+
+                char[] inputStringChars= inputString.ToUpper().ToCharArray();
+                Array.Sort(inputStringChars);
+                String sortedInputString = new string(inputStringChars);
+
+                for (int m=0; m<totalWords; m++)
+                {
+                    int inputStringLength = inputString.Length;
+                    int comparedWordLength = wordList[m].Length;
+
+                    if (inputStringLength == comparedWordLength)
+                    {
+                        char[] comparedWordChars = wordList[m].ToUpper().ToCharArray();
+                        Array.Sort(comparedWordChars);
+                        String sortedComparedWord = new String(comparedWordChars);
+                        if (sortedInputString == sortedComparedWord) 
+                        {
+                            matchedWordsListFinal.Add(wordList[m]);
+                            richTextBox2.Lines = matchedWordsListFinal.ToArray();
+                        }
+                    }
+                     else
```

Comment:
for -> else -> for -> if -> if
This nesting breaks readability. Consider the following code complexity rule: for/while = 2, if/else = 1. 
!!! Do NOT allow a complexity more 3-4 per method otherwise the code becomes unreadable. 
How can you fix this?